integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/AsyncPublisherAutoRetryTest.java

```diff
@@ -0,0 +1,74 @@
+package pl.allegro.tech.hermes.integrationtests;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.web.reactive.function.client.WebClient;
+import pl.allegro.tech.hermes.client.HermesClient;
+import pl.allegro.tech.hermes.client.HermesClientBuilder;
+import pl.allegro.tech.hermes.client.HermesSender;
+import pl.allegro.tech.hermes.client.webclient.WebClientHermesSender;
+import pl.allegro.tech.hermes.test.helper.util.Ports;
+
+import java.net.URI;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+public class AsyncPublisherAutoRetryTest {
```

Comment:
What do you think about moving this test to the hermes-client module? Take a look at `pl.allegro.tech.hermes.client.ReactiveHermesSenderTest`. This one can also be considered an integration test but is placed in the hermes-client module. 

The other way around approach is also possible, of course, but I lean towards having only tests in the integration-tests module that span multiple hermes modules.

Reply:
Looking more into this I wonder if this test is needed at all. In `hermes-client` module we already have tests covering cases tested here. I'd vote for removing this test class completely